### PR TITLE
Build the Clippy book as part of x.py doc

### PR DESCRIPTION
**Pull request by flip1995**
r? @ehuss since you said you would be interested in helping moving this forward.

cc @jyn514 as part of the bootstrap team.

### CODE REVIEW

File: src/bootstrap/doc.rs [hunk 1]
```diff
@@ -74,6 +74,7 @@ macro_rules! book {
 // and checking against it?).
 book!(
     CargoBook, "src/tools/cargo/src/doc", "cargo", submodule = "src/tools/cargo";
+    ClippyBook, "src/tools/clippy/book", "clippy", submodule = "src/tools/clippy";
```
Comment (jyn514) [inline src/bootstrap/doc.rs:77]: not a submodule, you can remove that bit.
```suggestion
    ClippyBook, "src/tools/clippy/book", "clippy";
```

**Review by jyn514 — APPROVED** [review]

**Comment by flip1995** [comment]
This now passes CI and is ready for review.

**Comment by ehuss** [comment]
@bors r-

Can you include clippy in the index page?  Possibly somewhere around here: https://github.com/rust-lang/rust/blob/d2074cbeeceb8b28fb6249a9aa3c937fe1c0bb92/src/doc/index.md?plain=1#L95

The changes to rename the `README.md` files shouldn't be necessary.  mdbook does translation of README.md to index.html.  You can keep the changes if you really want. One benefit with using README is so that the source can be easily browsable on GitHub's source viewer (like https://github.com/rust-lang/rust-clippy/tree/master/book/src). That's not too important I think, so it's up to you.

**Comment by flip1995** [comment]
When using `README.md` as the file name, linking to to directory/index file doesn't work, see: https://github.com/rust-lang/rust/pull/98848#issuecomment-1173501071

```
Building stage0 tool linkchecker (x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.14s
clippy/print.html:216: broken link - `clippy/continuous_integration/README.html`
clippy/usage.html:176: broken link - `clippy/continuous_integration/README.html`
```

When trying to create a link to just `continuous_integration/` it complains, that `index.html` should be used. When using `continuous_integration/README.md` to link it, I get the above error. 

If you have an idea how to resolve this, please let me know! I'd also prefer the `README.md` file names, but adapted the `index.md` file names like they are named in the cargo book.

> Can you include clippy in the index page? Possibly somewhere around here:

WIll do.

**Comment by ehuss** [comment]
I think if you rename the `index.md` files back to `README.md` as-is in your current PR, and change `SUMMARY.md` to link to `README.md`, everything should be fine.

It is a known bug that the readme linking is kinda wonky (https://github.com/rust-lang/mdBook/issues/984).  The SUMMARY.md file should be `README.md`, but the links between chapters inside the content need to either be `index.md` or `index.html`.  The linkifier doesn't know about the README translation.  I wish that could get fixed.

**Comment by flip1995** [comment]
This should be done now.

**Pull request merged**